src/main/java/uk/gov/hmcts/reform/jps/services/refdata/LocationService.java

```diff
@@ -18,6 +18,33 @@
 public class LocationService {
     private final LocationServiceClient regionServiceClient;
 
+    public Optional<CourtVenue> getCourtVenue(String hmctsServiceCode, String epimmsId) {
+        return getCourtVenue(
+            getLocationApiResponse(hmctsServiceCode),
+            epimmsId,
+            (court, epimsId) -> court.getEpimmsId().equals(epimsId)
+        );
+    }
+
+    private Optional<CourtVenue> getCourtVenue(LocationApiResponse serviceCourtInfo,
+                                               String value,
+                                               BiPredicate<CourtVenue, String> predicate) {
+        if (null == serviceCourtInfo) {
+            return Optional.empty();
+        }
+        return serviceCourtInfo.getCourtVenues().stream()
+            .filter(courtVenue -> predicate.test(courtVenue, value))
+            .findAny();
+    }
+
+    public String getVenueName(String hmctsServiceCode, String epimmsId) {
+        Optional<CourtVenue> courtVenue = getCourtVenue(hmctsServiceCode, epimmsId);
```

Comment:
This could be changed to
return getCourtVenue(hmctsServiceCode, epimmsId)
            .map(CourtVenue::getVenueName)
            .orElse("");

Reply:
LGTM